### PR TITLE
Chore: Migrate to Expo SDK 56 CNG (Android-only)

### LINKED ISSUE #109: Env audit + typed runtime validation
> **Scope note:** This ticket is **verification/validation only, NOT a rewrite**. The env migration (`@env`/`react-native-dotenv` removed → `EXPO_PUBLIC_*` via `process.env`, `apiBaseUrl.ts`, `env.d.ts`) was **already done on main**. Remaining scope: **confirm no residual `@env` references + optionally add typed runtime validation**. Assigned to @mattrace-gloo.

### Current state in baseline

- `src/config/apiBaseUrl.ts` reads `process.env.EXPO_PUBLIC_API_BASE_URL` and throws if unset.
- `src/types/env.d.ts` declares `EXPO_PUBLIC_API_BASE_URL: string` under `NodeJS.ProcessEnv`.
- `.env.example` = `EXPO_PUBLIC_API_BASE_URL=http://10.0.2.2:9999`; `jest.env.cjs` defaults `http://localhost:9999`.
- Test CI (`.github/workflows/test.yml`) sets `EXPO_PUBLIC_API_BASE_URL: http://localhost:9999`. No `FLUENT_USER_EMAIL` anywhere.
- `@env`/`react-native-dotenv` are **gone from main** (note: [PR #82](https://github.com/eten-tech-foundation/fluent-mobile/pull/82) still imports `@env` in `recordingSync.ts` — an anti-pattern, not merging).

### Problems / why

- Validation is single-key and ad-hoc (only the base URL, only at first use). New env (`EXPO_PUBLIC_OFFLINE_MODE` from #107) will need typing/validation too.
- No central typed config object; callers read `process.env` in scattered spots.
- Need a guard to prevent `@env` regressions creeping back (coordinate with lint #114).

### Target design

- A single typed config module (e.g. `src/config/env.ts`) that reads and validates all `EXPO_PUBLIC_*` values once and exports a typed object:

```ts
export const env = {
 apiBaseUrl: requireUrl('EXPO_PUBLIC_API_BASE_URL'), // optional in offline mode
 offlineMode: boolEnv('EXPO_PUBLIC_OFFLINE_MODE'),
} as const;
```

- Keep `getApiBaseUrl()` (back it with the new module) so existing imports keep working; throw with the same actionable message.
- Extend `src/types/env.d.ts` with any new keys (`EXPO_PUBLIC_OFFLINE_MODE`).
- Document the env contract in onboarding docs (#112).

### Implementation steps

1. Grep the repo for any residual `@env` / `react-native-dotenv` references; remove if found (none expected on main).
2. Add `src/config/env.ts` with typed accessors + validation; re-implement `getApiBaseUrl()` on top of it.
3. Add `EXPO_PUBLIC_OFFLINE_MODE` (and any others) to `env.d.ts`.
4. Document env vars in docs; ensure `.env.example` is complete.
5. Coordinate a lint guardrail (#114) banning direct `process.env` reads outside `src/config/`.
6. Unit-test validation (missing/blank URL throws actionable error; offline flag parsing).

### Acceptance criteria (verify-only)

- [ ] **Confirm** no `@env`/`react-native-dotenv` references remain (migration already done on main).
- [ ] **Optional hardening:** a single typed config module validates all `EXPO_PUBLIC_*` keys; `getApiBaseUrl()` delegates to it.
- [ ] **Optional:** new keys typed in `env.d.ts`; `.env.example` complete.
- [ ] **Optional:** validation covered by unit tests.

### Test plan / verification

```bash
npm run format:check && npm run lint && npm run typecheck && npm test -- --ci
```

- Tests default `EXPO_PUBLIC_API_BASE_URL=http://localhost:9999` via `jest.env.cjs`; toggle env in-test to assert validation; **no live API required**.
- Do **not** use `FLUENT_USER_EMAIL`.

### Out of scope

- Offline-mode behavior (#107), lint rule implementation (#114), docs page authoring (#112).

### Links

- Epic: #120
- Related: #107 (offline flag), #114 (lint), #112 (docs)
- Anti-pattern: [PR #82](https://github.com/eten-tech-foundation/fluent-mobile/pull/82) `recordingSync.ts` `import { API_BASE_URL } from '@env'`

### LINKED ISSUE #122: Migrate to Expo SDK 56 CNG (Android-only)
## Summary

Migrate Fluent Mobile to **Expo SDK 56 Continuous Native Generation (CNG)**, Android-only. This replaces the previously committed native `android/` project with a managed, prebuild-driven workflow (`expo prebuild --clean --platform android`), and standardizes config + build tooling around Expo.

## What shipped

- **CNG / Expo SDK 56** config: `app.config.ts` (with `runtimeVersion: { policy: 'appVersion' }`), `eas.json`, EAS workflows under `.eas/`.
- **Env standardization:** removed `react-native-dotenv`/`@env` in favor of `EXPO_PUBLIC_*` via `process.env`; added `src/config/apiBaseUrl.ts` and `src/types/env.d.ts`.
- **Test stack:** migrated to `jest-expo` (`jest.config.cjs`, `jest.env.cjs`).
- **Theme + sync touch-ups** under `src/theme/*` and `src/services/sync.ts`.
- Preview-build workflow + dev-client onboarding docs.

## Attribution

Delivered by @mattrace-gloo in (merging to `main`).

## Status

**Completed** — Expo SDK 56 CNG migration merged to `main`. Tracking/credit issue.

### PR DESCRIPTION
**Pull request by mattrace-gloo**
## 🔥 TLDR
Migrates Fluent Mobile from bare React Native CLI to Expo SDK 56 with Continuous Native Generation (Android-only), removing committed native trees and adding EAS tag + PR preview release workflows.

## 📋 Summary
- **Issue**: The app used bare RN CLI with committed `android/`/`ios/`, `react-native-dotenv`, and no standardized EAS release or QA preview path.
- **Root Cause**: Pre-CNG workflow required maintaining generated native code in git and lacked automated preview/production pipelines aligned with other Expo apps.
- **Solution**: Adopt Expo SDK 56 + `app.config.ts`, gitignore generated native dirs, `expo prebuild --platform android` in CI, EAS tag/preview workflows, and `EXPO_PUBLIC_*` env vars.
- **Impact**: Android-only CNG reduces native drift; tag releases and `preview-build` PR labels enable QA and production builds without iOS scope.

**Related Issue:** chore/expo-cng-migration (no Linear/GitHub issue linked)

**Type of Change:**
- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [ ] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [x] 🔧 Maintenance/refactor (no functional changes)

---

### Key Files Modified

**`app.config.ts`** (new)
- Expo config with EAS project ID, `expo-build-properties`, bootsplash plugin, custom RNScreens fragment factory plugin
- `expo-updates` URL, `runtimeVersion`, `APP_VERSION` / `EAS_UPDATE_CHANNEL` env support

**`package.json` / tooling**
- Expo SDK 56, RN 0.85, `expo-dev-client`, `expo-updates`, `jest-expo`, `babel-preset-expo`
- Scripts: `expo start`, `expo run:android`, `npm run prebuild` (Android-only)
- Removed `react-native-dotenv`, stale RN CLI metro/babel presets

**`src/services/api.ts` / `connectivity.ts`**
    const API_BASE_URL = process.env.EXPO_PUBLIC_API_BASE_URL ?? '';

**Native removal + CNG**
- Deleted committed `android/`, `ios/`, `app.json`, `debug.keystore`
- `.gitignore`: `/android`, `/ios`, `.expo/`, Expo signing artifacts, `expo-env.d.ts`

**EAS / CI**
- `.github/workflows/eas-build.yml` — sync `app.config.ts` version on `v*` tags
- `.github/workflows/preview-build.yml` — `preview-build` label → OTA or Android APK
- `.eas/workflows/create-production-builds.yml` — Android production build + submit
- `.github/workflows/build.yml` — `expo prebuild --platform android` before Gradle

**`eas.json`**
- `development` / `preview` / `production` Android profiles; preview channel; production `autoIncrement` + Play internal submit track

**`assets/`**
- App icon, adaptive icon, Android bootsplash source assets

**Lines changed**: ~23k lines touched (mostly `package-lock.json`) | **Files modified**: 92

### Dependencies & Configuration
- [ ] No new dependencies
- [x] New dependencies: `expo`, `expo-dev-client`, `expo-updates`, `expo-build-properties`, `babel-preset-expo`, `jest-expo`, `eas-cli` (dev)
- [ ] No config changes
- [x] Environment/build config changes: `.env.example` uses `EXPO_PUBLIC_API_BASE_URL`; removed `FLUENT_USER_EMAIL` from CI/docs; `EXPO_TOKEN` required for EAS workflows
- [x] No database changes
- [ ] Database/storage changes:

---

### Testing Completed
- [ ] ✅ Android device/emulator tested
- [x] ✅ Unit tests added/updated and passing
- [x] ✅ Edge cases considered and tested

### Code Quality (fluent-mobile gates)
- [x] ✅ `npm run lint`
- [x] ✅ `npm run format:check`
- [x] ✅ `npm run typecheck`
- [x] ✅ `npm test -- --ci`
- [x] ✅ Self-reviewed the code changes

---

## 📸 Screenshots/Recordings
N/A — infrastructure/tooling migration; no UI feature changes (minor `PageHeader` StyleSheet API fix only).

---

## 🎯 Why This Solution?
1. **CNG alignment**: Native `android/` is generated from `app.config.ts` + plugins — matches Expo best practice and removes committed native drift.
2. **Android-only scope**: No iOS app, builds, or assets; simplifies CI and EAS workflows permanently.
3. **Release parity**: Tag + preview workflows mirror proven patterns from sibling Expo repos, adapted for Fluent’s semver and dev client.

## 📱 Before/After
- **Before**: Bare RN CLI, committed `android/`/`ios/`, `react-native start` / `run-android`, `@env` dotenv, manual EAS builds.
- **After**: Expo dev client, `npm run prebuild` + `npm run android`, `EXPO_PUBLIC_*` env, automated tag releases and PR preview OTA/APK.

---

## ⚠️ Breaking Changes
- [ ] No breaking changes
- [x] Breaking changes documented below:

- **Local setup**: Developers must run `npm run prebuild` before first `npm run android`; `.env` key renamed to `EXPO_PUBLIC_API_BASE_URL`.
- **Committed native code removed**: Any workflow editing `android/` directly must move to `app.config.ts` / `plugins/`.
- **iOS project removed** from repo (Android-only permanently).
- **CI**: Android build job runs `expo prebuild` first.

---

## 📊 Performance Impact

**Bundle Size:**
- [ ] No significant impact (< 1MB)
- [x] Minor increase (1-5MB)
- [ ] Significant increase (> 5MB) - justified because:

**Runtime Performance:**
- [x] No performance impact
- [ ] Performance improvements
- [ ] Potential performance concern - mitigated by:

**Battery/Memory:**
- [x] No impact on battery or memory usage
- [ ] Optimizations that improve battery/memory
- [ ] Potential impact - acceptable because:

---

## 🧪 How to Test

**Prerequisites:** Node 24+, `npm install`, copy `.env.example` → `.env`, set `EXPO_PUBLIC_API_BASE_URL`

**Steps:**
1. `npm run format:check && npm run lint && npm run typecheck && npm test -- --ci`
2. `npx expo prebuild --platform android --no-install`
3. `cd android && ./gradlew assembleDebug --no-daemon`
4. Locally: `npm start` + `npm run android` — confirm app launches and syncs
5. Confirm PR diff excludes `.env`, `.expo/`, and generated `android/`/`ios/` directories

**Expected:** CI gates pass; debug APK builds after prebuild; app runs on Android emulator/device with dev client.

---

## 🔗 Additional Context

- EAS project ID: `b0919574-f268-4768-b3bd-7cfa5172bbab`
- Docs: `.eas/README.md`, `.github/README.md`, `docs/AGENT_ONBOARDING.md`, `.cursor/rules/android-only.mdc`

**Deployment Notes:**
- Add `EXPO_TOKEN` to GitHub Actions secrets before tag/preview workflows work
- Create `preview-build` label for PR preview workflow
- Configure Google Play credentials in EAS for production submit (`submit: true`)

**Follow-up Tasks:**
- [ ] Smoke-test `preview-build` label on this PR after `EXPO_TOKEN` is set
- [ ] Push first version tag (e.g. `v1.0.0`) to validate production EAS workflow
- [ ] Mock or gate `fluent-api.test.ts` for offline CI (pre-existing network dependency)

---

## 🔗 Linked issues

Closes #122
Closes #109

### CODE REVIEW

**Comment by mattrace-gloo** [comment]
@chadw-eten

**Comment by mattrace-gloo** [comment]
Heads up for QA testing preview QR builds

You might hit a solid blue screen after installing the Fluent preview app from the PR link instead of seeing the login screen.

This is expected for this build right now. The PR is still in review, and it depends on a separate chapter-assignment / database query fix that hasn’t been merged yet. Because of that, the app can install and launch successfully but fail silently on a real device.

This isn’t an issue with your phone or the QR install steps — it’s just a known gap between two in-flight PRs.

For now:
- Don’t spend time deep-testing login or My Work flows on this build.
- Retest once the chapter-assignment fix is merged and a new preview build is posted on the PR.
- If the new build still opens to a blue screen, drop the PR link you used and we’ll post a fresh install.

Reminder: use the Fluent preview app from the PR comment, not Expo Go.

File: .cursor/commands/create-pr.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ Run and report results:
 npm run format:check
 npm run lint
 npx tsc --noEmit
-FLUENT_USER_EMAIL=test@example.com npm test -- --ci
+npm test -- --ci
```
Comment (fel-cesar) [inline .cursor/commands/create-pr.md:27]: kudos: Thanks for this

File: .eas/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    ├─ Set `APP_VERSION_FALLBACK` in app.config.ts to match tag
+    ├─ Commit [skip ci] and move tag to that commit
+    └─ Push to main
+    ↓
+EAS Workflow (create-production-builds.yml)
+    ├─ Fingerprint check
+    └─ Android production build → submit (internal track)
+```
+
+## One-time setup
```
Comment (fel-cesar) [inline .eas/README.md:28]: kudos: Thanks for the readme here

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -20,33 +20,16 @@ jobs:
...
-  #       run: cd ios && pod install
-  #     - name: Build iOS
-  #       run: |
-  #         xcodebuild \
-  #           -workspace ios/Fluent.xcworkspace \
-  #           -scheme Fluent \
-  #           -configuration Debug \
-  #           -sdk iphonesimulator \
-  #           CODE_SIGNING_ALLOWED=NO \
-  #           build
```
Comment (fel-cesar) [inline .github/workflows/build.yml:52]: kudos: thanks for removing this for now

File: android/app/src/main/res/values/styles.xml
Comment (fel-cesar) [inline android/app/src/main/res/values/styles.xml:1]: question(non-blocking): Why did we removed these android assets? Are they managed by expo now? Or somewhere else?

File: docs/guides/qa-preview-testing.md
Comment (fel-cesar) [inline docs/guides/qa-preview-testing.md:1]: kudos: thanks for the guides

File: plugins/withRNScreensFragmentFactory.js
Comment (fel-cesar) [inline plugins/withRNScreensFragmentFactory.js:1]: comment: No idea what this does for now.

File: app.config.ts
Comment (fel-cesar) [inline app.config.ts:1]: 👀

File: index.js [hunk 4]
```diff
@@ -1,9 +1,9 @@
-/**
- * @format
- */
+// Dev client native code exists only in the `development` EAS profile — not preview/production.
+if (__DEV__) {
+  require('expo-dev-client');
+}
```
Comment (fel-cesar) [inline index.js:4]: comment: Ai raised some ordering concerns regarding this, but I think we can move forward and come back if this becomes an issue

File: jest.config.cjs [hunk 5]
```diff
@@ -1,7 +1,5 @@
 module.exports = {
-  preset: 'react-native',
-  transformIgnorePatterns: [
-    'node_modules/(?!(@react-native|react-native|@react-navigation|react-native-safe-area-context|@react-native-vector-icons|@react-native-community/netinfo|react-native-screens|react-native-gesture-handler|react-native-svg|lucide-react-native)/)',
-  ],
-    testPathIgnorePatterns: ['/node_modules/', '/android/', '/ios/'],
+  preset: 'jest-expo',
```
Comment (fel-cesar) [inline jest.config.cjs:2]: Didnt know that we needed an specific jest config for expo.

**Review by fel-cesar — DISMISSED** [review]
+1 - Awesome work!

Comments ( non blocking ):

* I have some concerns regarding version bumping and GHA order of executing. As far as I understood bumping and publishing are using the same trigger "v*". But this is non critical as we dont have prod right now and we can check that empirically during development cycle.

* I also dont have much knowledge on the new EXPO env and the CNG, so I cant actually evaluate the functionality and the pertinece of the code so far. But I see no smells or bad practices in the reviewed parts.

thanks for updating the mdc files as well

File: .cursor/commands/create-pr.md (hunk 1 above)
Comment (mattrace-buildmidwestern) [reply .cursor/commands/create-pr.md:27]: Thanks! 🙌 Same PR template + gate commands we use in CI — just wired into a slash command so we don't re-type it each time.

File: .eas/README.md (hunk 2 above)
Comment (mattrace-buildmidwestern) [reply .eas/README.md:28]: Appreciate it — wanted a short map of the EAS side since most of the release flow lives outside the app code now.

File: .github/workflows/build.yml (hunk 3 above)
Comment (mattrace-buildmidwestern) [reply .github/workflows/build.yml:52]: Yep — removed the commented-out iOS job. We're **Android-only** for now, and with CNG there's no committed `ios/` folder anyway. Same idea as dropping a platform CI job you're not shipping yet — less noise until we actually need it.

File: android/app/src/main/res/values/styles.xml
Comment (mattrace-buildmidwestern) [reply android/app/src/main/res/values/styles.xml:1]: Good question — yes, Expo manages them now via **CNG** (Continuous Native Generation).

We stopped committing `android/` (and `ios/`). Instead, `app.config.ts` + config plugins are the source of truth, and `expo prebuild` generates the native project locally or in CI/EAS — similar to keeping `pubspec.yaml` + platform config as source of truth and regenerating native folders at build time rather than checking them in.

Things like splash, theme, and package name come from:
- `app.config.ts` (app id, icons, SDK levels)
- plugins (`expo-build-properties`, `react-native-bootsplash`, etc.)
- generated output under `android/` — gitignored, recreated on demand

File: plugins/withRNScreensFragmentFactory.js
Comment (mattrace-buildmidwestern) [reply plugins/withRNScreensFragmentFactory.js:1]: This is an **Expo config plugin** — a small build-time patch that runs during `expo prebuild`.

`react-native-screens` needs a line in `MainActivity.onCreate` to set a custom `FragmentFactory` (Android fragment restoration). Expo's generated `MainActivity` doesn't include it by default, so this plugin injects it automatically — like when a package doc says "add this to MainActivity," except it's codified so every prebuild/EAS build gets the same patch without manual edits.

File: app.config.ts
Comment (mattrace-buildmidwestern) [reply app.config.ts:1]: This is the central config for the whole app — one typed file instead of scattered native files.

Rough mapping if you're used to a single project manifest:
- name, version, assets → top-level fields here
- AndroidManifest / Gradle settings → `android` block + `expo-build-properties` plugin
- flavors / build variants → EAS profiles in `eas.json`
- native build hooks → `plugins: [...]`

`APP_VERSION_FALLBACK` is what release CI bumps to match `v*` tags before EAS builds.

File: index.js (hunk 4 above)
Comment (mattrace-buildmidwestern) [reply index.js:4]: Agreed — fine to ship and revisit if needed.

Quick context on the two lines:

1. `expo-dev-client` — only loaded in `__DEV__`; gives us a custom dev client for native modules. Not used in preview/production builds.

2. `react-native-gesture-handler` import — older RN versions required this as the **first** import. With RN 0.85 + `GestureHandlerRootView` in `App.tsx`, it's mostly legacy insurance. CodeRabbit flagged it; we kept it harmlessly for now — a bit like keeping an old `ensureInitialized()` call that's probably redundant on newer SDKs but low risk.

If we see gesture weirdness on Android, this is the first place I'd look.

File: jest.config.cjs (hunk 5 above)
Comment (mattrace-buildmidwestern) [reply jest.config.cjs:2]: Expo ships **`jest-expo`** — a Jest preset that mocks RN/Expo native modules so tests run in Node without a device or emulator, similar to how the test runner provides framework bindings so you don't need a full engine for every unit test.

Our config is minimal — preset + `jest.env.cjs` for env vars. Same tests as before, just compatible with Expo's module graph.

**Review by mattrace-buildmidwestern — APPROVED** [review]

**Review by fel-cesar — APPROVED** [review]
+1 on rebase
